tests/js/client/shell/shell-debug-dump.js

```diff
@@ -0,0 +1,345 @@
+      explainer.debugDump(fileName, query, {}, {examples: 10});
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+      let vn1After = [];
+      db[vn1].all().toArray().forEach(vertex => {
+        vn1After.push(vertex._key);
+      });
+      vn1After.sort();
+      let vn2After = [];
+      db[vn2].all().toArray().forEach(vertex => {
+        vn2After.push(vertex._key);
+      });
+      vn2After.sort();
+      let relationAfter = [];
+      db[relationName].all().toArray().forEach(edge => {
+        relationAfter.push(edge._key);
+      });
+      relationAfter.sort();
+      assertEqual(vn1Before, vn1After);
+      assertEqual(vn2Before, vn2After);
+      assertEqual(relationBefore, relationAfter);
+    },
+  };
+}
+
+function debugDumpGraphTestsuite() {
+  return {
+    setUp: function () {
+      db._createDatabase(dbName);
+      db._useDatabase(dbName);
+      db._create(vn1);
+      db._create(vn2);
+      db._createEdgeCollection(cn1);
+      db._createEdgeCollection(cn2);
+      for (let i = 0; i < 10; ++i) {
+        db[vn1].save({_key: `value-${i}`, value: i, type: 'value'});
+        db[vn2].save({_key: `test-${i}`, test: i, type: 'test'});
+      }
+      for (let i = 0; i < 10; ++i) {
+        db[cn1].save(`${vn1}/value-${i}`, `${vn2}/test-${i}`, {type: "is-in"});
+        if (i < 9) {
+          db[cn2].save(`${vn1}/value-${i}`, `${vn1}/value-${i + 1}`, {type: "is-in"});
+        }
+      }
+      const rel1 = graphModule._relation(cn1, vn1, vn2);
+      const rel2 = graphModule._relation(cn2, vn1, vn1);
+      graphModule._create(graphName, [rel1, rel2], null);
+    },
+
+    tearDown: function () {
+      db._useDatabase("_system");
+      db._dropDatabase(dbName);
+      try {
+        fs.unlink(fileName);
+      } catch (err) {
+      }
+      try {
+        fs.unlink(outFileName);
+      } catch (err) {
+      }
+    },
+
+    testDebugDumpWithGraphNamed: function () {
+      const graphBefore = db._graphs.document(graphName);
+      delete graphBefore._rev;
+      const fileName = fs.getTempFile() + "-debugDump";
+      const query = `FOR v,e, p IN 0..10 ANY  '${vn1}/value-1' GRAPH 'myGraph'  RETURN {vertex: p}`;
+      explainer.debugDump(fileName, query, {}, {});
+      const outFileName = fs.getTempFile() + "-inspectDump";
+      explainer.inspectDump(fileName, outFileName);
+      recreateEmptyDatabase();
+      internal.load(outFileName);
+      const graphAfter = db._graphs.document(graphName);
+      delete graphAfter._rev;
+      assertEqual(graphBefore, graphAfter, "graph recreated after debugDump is not the same as the original");
+      assertNull(db[vn1].any());
```

Comment:
Why are the underlying collections actually empty here? Is it because we did not specify `example: x` when creating the debug dump?

Reply:
Yes